### ISSUE TITLE
support non-standard EmojiReact activity type?

### ISSUE THREAD

**Issue by snarfed** [snarfed/bridgy-fed#484]
Started seeing these from https://mostr.pub/ recently, evidently Pleroma pioneered these and other services now maybe support them too?

- https://docs-develop.pleroma.social/backend/development/API/differences_in_mastoapi_responses/#emojireact-notification
- https://github.com/misskey-dev/misskey/issues/5425
- https://libervia.org/__b/doc/backend/components.html#reactions
- https://socialhub.activitypub.rocks/t/fediverse-enhancement-proposal-fep/1059/3
- https://socialhub.activitypub.rocks/t/vocabularies-vs-extension/3115

Example:

```json
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {
      "mostr": "http://mostr.pub/ns#",
      "Zap": "mostr:Zap",
      "nostr": "mostr:nostr"
    }
  ],
  "type": "EmojiReact",
  "id": "https://mostr.pub/objects/51f6e930c624468ac497f66c942c1c39b50cc76330e85c3ba4c0d407534f7163",
  "object": "https://mostr.pub/objects/1b7b03a9ae88ca28327376fda27be9f6447d98d26081cfb94fb87a8d5eeb400a",
  "content": "\ud83e\udd19",
  "actor": "https://mostr.pub/users/82341f882b6eabcd2ba7f1ef90aad961cf074af15b9ef44a09f9d2a8fbfbe6a2",
  "to": [
    "https://www.w3.org/ns/activitystreams#Public",
    "https://mostr.pub/users/32e1827635450ebb3c5a7d12c1f8e7b2b514439ac10a67eef3d9fd9c5c68e245"
  ],
  "cc": [
    "https://mostr.pub/users/82341f882b6eabcd2ba7f1ef90aad961cf074af15b9ef44a09f9d2a8fbfbe6a2/followers"
  ],
  "nostr": {
    "id": "51f6e930c624468ac497f66c942c1c39b50cc76330e85c3ba4c0d407534f7163",
    "pubkey": "82341f882b6eabcd2ba7f1ef90aad961cf074af15b9ef44a09f9d2a8fbfbe6a2"
  }
}
```

**Comment by Tamschi** [comment]
["Zaps" are payments](https://nostr.how/en/zaps), so it's not quite an emoji reaction, the bridge just renders it as such.

I believe you can't send them to non-nostr users either. If I remember correctly, the receiving side has to respond with an invoice on Nostr before the payment completes, which the bridge hopefully doesn't implement.

**Comment by snarfed** [comment]
Agreed! Microtransactions are intriguing, but not something I plan to try to bridge. I was only really thinking about the emoji reaction part here, so this zap probably isn't the best example, we should replace it with a native fediverse example, maybe from Pleroma.

**Comment by Tamschi** [comment]
I think the are currently no supported protocols other than ActivityPub that transport emoji reactions, but I assume forwarding them to a webhook(?) is in scope for websites bridged there.

**Comment by snarfed** [comment]
The IndieWeb stack supports emoji reactions: https://indieweb.org/reacji . Bluesky doesn't though, you're right. We could reasonably translate them as replies with the emoji as text, or just ignore them.

(For the future, Nostr emoji reactions are https://github.com/nostr-protocol/nips/blob/master/25.md, and custom emoji are https://github.com/nostr-protocol/nips/blob/master/30.md. Haven't looked at other protocols yet.)

**Comment by silverpill** [comment]
ActivityPub emoji reactions: https://codeberg.org/fediverse/fep/src/branch/main/fep/c0e0/fep-c0e0.md
